R/is.R

```diff
@@ -1,318 +0,0 @@
-#' Is a named argument
```

Comment:
I am not sure if we can remove this file completely. `is_named()`, `is_auto`, `is_date()`, and `is_valid_dtc()` are called in admiral.

Reply:
I agree that some of the `is_*` functions might need to stay, so far I kept (and moved) `is_order_vars()` to the `R/dev_utilities.R`.  

- we can use `assert_date_vector()` to replace `is_date()`
- `{admiral}` only call `is_named()` in 2 locations, are there any reservations to `(!is.null(names(arg)) && all(names(arg) != "")` in those 2 locations, or should we just keep `is_named`?
![is_named](https://user-images.githubusercontent.com/112789549/200595927-c2a68912-c42b-4332-85f9-43013b1fe15a.png)

- `is_auto` is used in multiple `{admiral}` section of the `R/create_query_data.R`
- I'm also not against keeping `is_valid_dtc()` so what do you think @bundfussr @bms63 @thomas-neitmann 

All in all, if we do keep a few of the `is_*` functions, are we all okay to move them to the `R/dev_utilities.R` program as part of utility functions. So the functions I'm referring to are `is_order_vars()` (which I moved already), `is_valid_dtc()` and `is_auto()`. Please let me know what you think.

Reply:
- `is_date()` is used in `convert_date_to_dtm()`. We can not use `assert_date_vector()` there because no error should be thrown if it isn't a date.
- I would keep `is_named()` and as it is easier to read.
- I would also keep `is_valid_dtc()` as it is hard to replace.
- If we keep more than one of the `is_*()` functions.,I would not move them to `dev_utilities.R` but keep `is.R`. Then they are easier to find.
@bms63 , @thomas-neitmann , what do you think?